pkg/apis/core/v1/conversion.go

```diff
@@ -306,6 +307,9 @@ func Convert_core_PodSpec_To_v1_PodSpec(in *core.PodSpec, out *v1.PodSpec, s con
 		out.HostUsers = in.SecurityContext.HostUsers
 	}
 
```

Comment:
This change will affect all objects that contain PodSpec (like deployments, replicasets, etc).

Can we scope this change to only pod objects (move this to Convert_v1_Pod_To_core_Pod / Convert_core_Pod_To_v1_Pod)

For non-pod objects, adding a warning to warningsForPodSpecAndMeta for negative TerminationGracePeriodSeconds (like "negative spec.terminationGracePeriodSeconds values are invalid and will be treated as a value of 1") would avoid breaking those controllers while still surfacing the issue with their request or manifest

Reply:
Along with those changes, we need test coverage of behavior when creating and updating a pod

Reply:
Done